pyccel/ast/core.py

```diff
@@ -232,7 +232,7 @@ def __init__(self, *args):
 
         if self.stage == 'syntactic':
             args = handle_precedence(args, self.precedence, self._commutative)
-            self._args = tuple(args)
+            self._args = args
```

Comment:
I would like to make a recommendation the about coding style. When possible we should try to avoid redefining a variable that already exists (as it happens here for `args`) and create a new variable instead. Here the obvious solution actually makes the code one line shorter:
```
self._args = handle_precedence(args, self.precedence, self._commutative)
```

Reply:
You are right. This came from laziness on my part originally (I didn't want to change `args` everywhere in the `__init__`) but as I have limited this step to the syntactic stage I don't need to change everything, just that line